lib/elixir/lib/inspect/algebra.ex

```diff
@@ -354,28 +347,24 @@ defmodule Inspect.Algebra do
             try do
               Process.put(:inspect_trap, true)
 
-              res =
-                Inspect.Map.inspect(struct, %{
-                  opts
-                  | syntax_colors: [],
-                    inspect_fun: Inspect.Opts.default_inspect_fun()
-                })
-
-              res = IO.iodata_to_binary(format(res, :infinity))
-
-              message =
-                "got #{inspect(caught_exception.__struct__)} with message " <>
-                  "#{inspect(Exception.message(caught_exception))} while inspecting #{res}"
-
-              exception = Inspect.Error.exception(message: message)
```

Comment:
We were defining in `Algebra` the way the `Inspect.Error` message was formatted and its contents, so I moved it to exception.ex and added the needed fields to the exception to format this message, these being `:exception`, `:stacktrace` and `:struct`.